crates/re_query/src/util.rs

```diff
@@ -57,21 +57,35 @@ impl VisibleHistory {
         to: VisibleHistoryBoundary::Infinite,
     };
 
-    pub fn from(&self, cursor: TimeInt) -> TimeInt {
+    /// Do not use this to build a [`TimeRange`], use [`Self::time_range`].
+    fn from(&self, cursor: TimeInt) -> TimeInt {
         match self.from {
             VisibleHistoryBoundary::Absolute(value) => TimeInt::from(value),
             VisibleHistoryBoundary::RelativeToTimeCursor(value) => cursor + TimeInt::from(value),
             VisibleHistoryBoundary::Infinite => TimeInt::MIN,
         }
     }
 
-    pub fn to(&self, cursor: TimeInt) -> TimeInt {
+    /// Do not use this to build a [`TimeRange`], use [`Self::time_range`].
+    fn to(&self, cursor: TimeInt) -> TimeInt {
         match self.to {
             VisibleHistoryBoundary::Absolute(value) => TimeInt::from(value),
             VisibleHistoryBoundary::RelativeToTimeCursor(value) => cursor + TimeInt::from(value),
             VisibleHistoryBoundary::Infinite => TimeInt::MAX,
         }
     }
+
+    pub fn time_range(&self, cursor: TimeInt) -> TimeRange {
+        let mut from = self.from(cursor);
+        let mut to = self.to(cursor);
+
+        // TODO(#4993): visible time range UI can yield inverted ranges
+        if from > to {
+            std::mem::swap(&mut from, &mut to);
+        }
```

Comment:
I think (but I'm not sure) that this is the source of a change of behaviour.

Before: moving the "from" boundary "against" the "to" boundary would push the latter, to forbid inverted range (at least in appearance). Now the UI outright allows inverted range, which is weird.

Reply:
Ha, nice catch; I'll see what I can do

Reply:
Alright, kept it simple; I'm pretty happy with it